### PR TITLE
LLVM: use @llvm.memset where possible

### PR DESCRIPTION
**Pull request by EgorBo**
TODO: make sure the following methods end up as `@llvm.memset` (for primitive types)

- [x] **RuntimeImports.ZeroMemory**
- [ ] **Unsafe.InitBlockUnaligned** (currently emits `string.memset` call)

E.g. `new Span<byte>(data).Clean()` calls `SpanHelpers.ClearWithoutReferences`  which either calls ZeroMemory or InitBlockUnaligned https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/SpanHelpers.cs#L29-L32  (@llvm.meset shows better performance results than both)

### CODE REVIEW

**Comment by filipnavara** [comment]
Issue #16716, for reference.

File: mono/mini/mini-ops.h [hunk 1]
```diff
@@ -768,6 +768,7 @@ MINI_OP(OP_SAVE_LMF, "save_lmf", NONE, NONE, NONE)
 MINI_OP(OP_RESTORE_LMF, "restore_lmf", NONE, NONE, NONE)
 
 MINI_OP3(OP_MEMMOVE, "memmove", NONE, IREG, IREG, IREG)
+MINI_OP3(OP_MEMSET64, "memset64", NONE, IREG, IREG, IREG)
```
Comment (vargaz) [inline mono/mini/mini-ops.h]: The last one should be LREG.

**Comment by vargaz** [comment]
Looks ok but some issues:
- gc polling for large sizes ?
- llvm might end up calling memset without a wrapper which can causes problems with unwinding/stack walks. This is one of the reasons we called a managed method.

**Comment by akoeplinger** [comment]
Should we port this to dotnet/runtime? I think we probably don't need it in mono/mono.

**Comment by EgorBo** [comment]
> Should we port this to dotnet/runtime? I think we probably don't need it in mono/mono.

Ah I don't think so, Zoltan mentioned possible issues with this so it's probably better to leave as is to avoid long gc pauses in some cases.

**Pull request closed without merging**